langserver/references.go

```diff
@@ -8,19 +8,30 @@ import (
 	"go/parser"
 	"go/token"
 	"go/types"
+	"log"
 	"path"
+	"runtime"
 	"sort"
 	"strings"
 	"sync"
 
 	"golang.org/x/tools/go/loader"
 	"golang.org/x/tools/refactor/importgraph"
 
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/sourcegraph/go-langserver/pkg/lsp"
 	"github.com/sourcegraph/jsonrpc2"
 )
 
 func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn JSONRPC2Conn, req *jsonrpc2.Request, params lsp.ReferenceParams) ([]lsp.Location, error) {
+	// TODO: Add support for the cancelRequest LSP method instead of using
+	// hard-coded timeouts like this here.
+	//
+	// See: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#cancelRequest
+	// We just use the same timeout as workspace/xreferences
+	ctx, cancel := context.WithTimeout(ctx, workspaceReferencesTimeout)
```

Comment:
This shouldn't be `workspaceReferencesTimeout` but rather something else. Otherwise, rename that var to `referencesTimeout`